lib/fluent/plugin/out_honeycomb.rb

```diff
@@ -72,34 +73,44 @@ def write(chunk)
           record.merge!(flatten(record[k], k))
           record.delete(k)
         end
+
+        if (@dataset_from_key != "" && record.has_key?(@dataset_from_key))
+          dataset = record[@dataset_from_key]
+          record.delete @dataset_from_key
+        else
+          dataset = @dataset
+        end
+        batch = batches[dataset]
         batch.push({
             "data" => record,
             "samplerate" => @sample_rate,
             "time" => Time.at(time).utc.to_datetime.rfc3339
         })
       end
 
-      publish_batch(batch, 0)
+      batches.each do |dataset, batch|
+        publish_batch(dataset, batch, 0)
+      end
     end
 
-    def publish_batch(batch, retry_count)
+    def publish_batch(dataset, batch, retry_count)
       if batch.length == 0
         return
       end
-      log.info "publishing #{batch.length} records"
+      log.info "publishing #{batch.length} records to dataset #{dataset}"
       body = JSON.dump(batch)
       resp = HTTP.headers(
           "User-Agent" => "fluent-plugin-honeycomb",
```

Comment:
unrelated to this change, but since we're here, could you add a version string to the user agent? (in the format `fluent-plugin-honeycomb/version`)